conan/tools/gnu/autotoolsgen.py

```diff
@@ -0,0 +1,35 @@
+import platform
+
+from conan.tools.env import VirtualEnv
+from conan.tools.gnu import AutotoolsToolchain, AutotoolsDeps
+
+
+class AutotoolsGen:
+    def __init__(self, conanfile):
+        self.toolchain = AutotoolsToolchain(conanfile)
+        self.deps = AutotoolsDeps(conanfile)
+        self.env = VirtualEnv(conanfile)
+
+    def build_environment(self):
+        envtoolchain = self.toolchain.environment()
+        envdeps = self.deps.environment()
+        build_env = self.env.build_environment()
+        build_env.compose(envtoolchain)
+        build_env.compose(envdeps)
+        return build_env
+
+    def run_environment(self):
+        run_env = self.env.run_environment()
+        return run_env
+
+    def generate(self):
+        build_env = self.build_environment()
+        run_env = self.run_environment()
+        # FIXME: Use settings, not platform Not always defined :(
+        # os_ = self._conanfile.settings_build.get_safe("os")
+        if platform.system() == "Windows":
+            build_env.save_bat("conanbuildenv.bat")
```

Comment:
name is too generic? may overwrite other generators, if multiple are used together...

Reply:
``conanxxxx`` filenames are generated by Conan (probably we should document this more explicitly too).
This filename is automatically used by ``self.run()`` and is the same filename generated by ``VirtualEnv`` (to be automatically injected too). The idea is that this file is already the aggregation of the necessary build environments, in a single file, oriented also to make user life easier at the consumer flow. 

Ideas welcome, any suggestion of other filename?

Reply:
I am not sure if I got the idea. what if we add similar generator e.g. for premake, which will also need `VirtualEnv`.
and what if project uses both `autotoolsgen` and `premake` (e.g. it has to sub-directories with different build systems)
now both will try to create `conanbuildenv.bat` to override each other?

Reply:
``Autotools`` generators do not create ``conanbuildenv.bat``, ``AutotoolsGen`` that wraps both Autotools and VirtualEnv does. In the same spirit, if you want to combine Autotools and Premake in an ordered, structured way, you can do it directly in the recipe, aggregating environments before saving ``conanbuildenv.bat`` or creating a generator that aggregates them and using it.